packages/components/src/TimeInput.test.tsx

```diff
@@ -149,6 +151,64 @@ describe('select and type', () => {
 
     testSelectAndType(1, '000000', '00:00:00');
   });
+  it('handles backspace', () => {
+    // Replace selected section with fixed-width spaces
+    testSelectAndType(
+      0,
+      '{backspace}',
+      `${FIXED_WIDTH_SPACE}${FIXED_WIDTH_SPACE}:34:56`
+    );
+    testSelectAndType(
+      3,
+      '{backspace}',
+      `12:${FIXED_WIDTH_SPACE}${FIXED_WIDTH_SPACE}:56`
+    );
+
+    // Allow deleting digits from the end
+    testSelectAndType(9, '{backspace}', `12:34`);
+
+    // Add missing mask chars
+    testSelectAndType(9, '{backspace}{backspace}12', `12:31:2`);
+  });
+
+  it('trims trailing mask and spaces', () => {
+    const { unmount } = makeTimeInput();
+    const input: HTMLInputElement = screen.getByRole('textbox');
+
+    input.setSelectionRange(3, 3);
+
+    userEvent.type(input, '{backspace}');
+
+    input.setSelectionRange(9, 9);
+
+    userEvent.type(input, '{backspace}');
+
+    expect(input.value).toEqual(`12`);
+
+    input.setSelectionRange(1, 1);
+
+    userEvent.type(input, '{backspace}');
+
+    expect(input.value).toEqual(``);
+
+    unmount();
+  });
+
+  it('existing invalid behaviors that might need to be fixed', () => {
+    // Expected: '20:34:56'?
+    testSelectAndType(1, '5{arrowleft}2', `25:34:56`);
```

Comment:
As mentioned above I think `20:34:56` makes more sense:
- Start with `12:34:56`, selection on the `2`
- Enter `5`, changes to `05:34:56`, selection on 3
- Press left, selection changes back to first segment
- Enter `2`, sees that `25` is not valid, so changes the second value to `20`
Honestly seems like such an edge case though that I'd almost rather not bother.